Makefile

```diff
@@ -177,24 +180,26 @@ docker-push-release:  docker-tag-release
 	@docker push $(REPOSITORY):$(VERSION)
 	@docker push $(REPOSITORY):latest
 
-# Build the docker image
-docker-build: test
-	docker build --pull . -t ${IMG}
+docker-buildx: test
+	@if ! docker buildx ls | grep -q container-builder; then\
+		docker buildx create --platform "linux/amd64,linux/arm64,linux/arm/v7" --name container-builder --use;\
```

Comment:
You should not need to specify platforms here.
When doing cross compilation the build platform is linux/amd64 no matter what the target platform is. This is specifying the build platform.

Reply:
Inside your Dockerfile you need to set `FROM --platform=$BUILDPLATFORM`, unless you want to do native compilation (instead of cross compile).

Reply:
@cpuguy83 if you look at the Dockerfile there is no need for adding `--platform=$BUILDPLATFORM` as mentioned in my previous reply, the final base is a distroless and we're simply coping the go binary in its compiled form to the final stage, there is no point adding that to the builder stage as golang doesn't require it in this context.

Reply:
buildkit will try to build using the platform specified as the target arch.
`FROM --platform=$BUILDPLATFORM` tells it to use the image for the build platform (That is the arch/os of the build host) instead of the target platform.

Reply:
This would go on your `builder` image, not the final.

Reply:
There is no value in putting it into the builder stage because go doesn't require the host OS to be multi-arch as long as the inputs are validated and correct which are passed from `TARGETPLATFORM` then it is sufficient as illustrated by this large community who maintain multi-arch images specifically for armv7l and arm64 https://github.com/raspbernetes/multi-arch-images .

Hope that makes sense? We don't need the builder image to be multiarch.

Reply:
The reason to do this is so you don't end up building in qemu, and instead build natively on the host arch with cross compilation.
Again, without setting the platform for the build image you will get qemu (or whatever is configured for binfmt), not cross compilation.